src/js/components/Menu/Menu.js

```diff
@@ -184,20 +184,40 @@ const Menu = forwardRef((props, ref) => {
     }
   };
 
-  const content = children || (
-    <Box
-      direction="row"
-      justify={justifyContent}
-      align="center"
-      pad={theme.button.default ? undefined : 'small'}
-      gap={label && icon !== false ? 'small' : undefined}
-    >
-      <Text size={size}>{label}</Text>
-      {icon !== false
-        ? (icon !== true && icon) || <MenuIcon color={iconColor} size={size} />
-        : null}
-    </Box>
-  );
+  const menuLabel = <Text size={size}>{label}</Text>;
+  const menuIcon =
+    icon !== false
+      ? (icon !== true && icon) || <MenuIcon color={iconColor} size={size} />
+      : null;
+
+  const isDefaultButton = theme.button.default && !children;
```

Comment:
This is a minor comment...

Can we use `let buttonProps;` and then assign it in the same way we assign `content`? That would have a single source for the conditional logic.